internal/application/workload.go

```diff
@@ -185,24 +156,53 @@ func (a *Workload) Replicas(ctx context.Context) (map[string]*models.PodInfo, er
 // Get returns the state of the app deployment encoded in the workload.
 func (a *Workload) Get(ctx context.Context) (*models.AppDeployment, error) {
 
-	deployment, err := a.Deployment(ctx)
+	// Information about the active workload.
+	// The data is pulled out of the list of Pods associated with the application.
+	// The originally asked the app's Deployment resource. With app charts the existence
+	// of such a resource cannot be guarantueed any longer.
+
+	podList, err := a.Pods(ctx)
 	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return nil, err
-		}
-		// App is inactive, no deployment, no workload
+		return nil, err
+	}
+
+	// No pods => no workload
+	if len(podList) == 0 {
 		return nil, nil
 	}
 
-	desiredReplicas := deployment.Status.Replicas
-	readyReplicas := deployment.Status.ReadyReplicas
+	var readyReplicas int32
+	var createdAt time.Time
+	var stageID string
+	var username string
+	var controllerName string
+
+	if len(podList) > 0 {
+		// Pods found. replace defaults with actual information.
+
+		// Initialize various pieces from the first pod ...
+		createdAt = podList[0].ObjectMeta.CreationTimestamp.Time
+		stageID = podList[0].ObjectMeta.Labels["epinio.io/stage-id"]
```

Comment:
This can also be non-true. When pushing an application again, the old Pods co-exist with the new ones for some time (usually for very little). After a while things balance out but I'm not sure this won't lead to wrong assumptions in some cases (e.g. getting the old stage id assuming it's the new one). To suggest a better alternative, we would have to find all consumers. Maybe we can get this information from the application itself (as part of this? https://github.com/epinio/epinio/issues/1657)

Reply:
I see, like `desiredReplicas`.

Reply:
yes